qcodes/dataset/data_set_cache.py

```diff
@@ -194,6 +196,20 @@ def to_xarray_dataset(self) -> xr.Dataset:
         data = self.data()
         return load_to_xarray_dataset(self._dataset, data)
 
+    def to_csv(
+        self,
+        path: str,
+        single_file: bool = False,
+        single_file_name: Optional[str] = None,
+    ) -> None:
+        dfdict = self.to_pandas_dataframe_dict()
+        dataframe_to_csv(
+            dfdict=dfdict,
+            path=path,
+            single_file=single_file,
+            single_file_name=single_file_name,
+        )
```

Comment:
I don't think we need / want this as a public method

Reply:
This was written because I confused the different kind of interfaces.

We have `to_xarray ...` and `to_pandas...` so it seemed to make sense to have `to_csv` but actually these are 2 different things. One is an in memory object type and the other one a file format. 

We have for now defined that the file formats should be accessed via export( export_type="csv") so it seems like we should just loose this method. We do have a `write_data_to_text_file` on the DataSet class which should perhaps be deprecated? 

The dataset has `write_data_to_text_file` so perhaps we should implement this. The